Kconfig.zephyr

```
@@ -486,6 +486,12 @@ config BUILD_OUTPUT_UF2_FAMILY_ID
 	  name string. If the SoC in use is known by UF2, the Family ID will
 	  be pre-filled with the known value.
 
+config BUILD_OUTPUT_UF2_IGNORE_FLASH_BASE
+	bool "Ignore configured flash base for UF2 offset calculation"
+
+config BUILD_OUTPUT_UF2_IGNORE_FLASH_OFFSET
+	bool "Ignore configured flash offset for UF2 offset calculation"
```

Maybe phrase it differently, like:
```suggestion
config BUILD_OUTPUT_UF2_FLASH_OFFSET
	bool "UF2 address offset adjusted"
	default y
	help
	  UF2 conversion will default be adjusted with the FLASH_LOAD_OFFSET value.
	  Disable this setting to use the FLASH_BASE_ADDRESS directly without adjusting for load offset.
```

That will make it default to use the load offset, which can then be disabled.

Note: this is not a blocker, just a proposal, so if you really prefer to keep setting this way feel free to do so.
However, if keeping current way I still ask you to provide some help text, and try to shorten the prompt, as it is very long.

---

I'm fine inverting it, but the naming seems like it might be the actual flash offset, not a bool. Perhaps `BUILD_OUTPUT_UF2_USE_FLASH_OFFSET`?

---

> Perhaps BUILD_OUTPUT_UF2_USE_FLASH_OFFSET?

fine with me.

Btw. does this setting require a prompt ?
Based on the talks in #34835, it could indicate this is something that is not for the user to decide, more a soc / board thing.

---

TBH... I don't think so? I can remove.

---

Updated.